File: algorithms/sorting/selection_sort/selection.dart
void main(){
  List<int> numbers = [412, 41513,223,90,2, 1];
  print("While unsorted, numbers are: ${numbers}");
  selection(numbers);
  print("After Sorting, numbers are: ${numbers}");

}



// Selection Sort
void selection(List _array) async {
  for (int i = 0; i < _array.length - 1; i++) {
    int minIdx = i;
    for (int j = i + 1; j < _array.length - 1; j++) {
      if (_array[minIdx] > _array[j]) {
        minIdx = j;
      }
    }
    int temp = _array[i];
    _array[i] = _array[minIdx];
    _array[minIdx] = temp;
  }
}

File: algorithms/sorting/selection_sort/selection.cpp
#include <iostream>
using namespace std;


int main() {
    
    int unsorted_arr[] = {15155, 124, 9824, 12, 12, 5};
    int length = sizeof(unsorted_arr) / sizeof(unsorted_arr[0]);
    
    cout << "UnSorted Arry Is: \n";
    for (int i =0; i < length; i++) {
        cout << unsorted_arr[i] << " ";
    }
    cout << "\n";
    
    
    
    // selection sort
    for (int i = 0; i < length - 1; i++) {
        for (int j = i + 1; j < length; j++){
            if (unsorted_arr[j] < unsorted_arr[i]) {
                int temp = unsorted_arr[j];
                unsorted_arr[j] = unsorted_arr[i];
                unsorted_arr[i] = temp;
            }
        }
    }
    cout << "Now Sorted Arry Is: \n";
    for (int i =0; i < length; i++) {
        cout << unsorted_arr[i] << " ";
    }
    cout << "\n";
}